lib/protocol/Writer.js

```diff
@@ -861,7 +861,7 @@ Writer.prototype[TypeCode.LONGDATE] = function writeLongDate(value) {
     hours = ~~ts[4];
     minutes = ~~ts[5];
     seconds = ~~ts[6];
-    nanoseconds = ~~((ts[6] * 1000000000) % 1000000000);
+    nanoseconds = ~~((ts[6] * 10000000) % 10000000);
```

Comment:
```suggestion
    nanoseconds = Number((ts[6]);
```

different alternative, more reliable.